torch/testing/_internal/common_utils.py

```diff
@@ -5365,11 +5366,20 @@ def remove_cpp_extensions_build_root():
     """
     Removes the default root folder under which extensions are built.
     """
-    default_build_root = torch.utils.cpp_extension.get_default_build_root()
+    default_build_root = cpp_extension.get_default_build_root()
     if os.path.exists(default_build_root):
         if IS_WINDOWS:
             # rmtree returns permission error: [WinError 5] Access is denied
             # on Windows, this is a workaround
             subprocess.run(["rm", "-rf", default_build_root], stdout=subprocess.PIPE)
         else:
             shutil.rmtree(default_build_root, ignore_errors=True)
+
+def scoped_load_inline(*args, **kwargs):
+    temp_dir = tempfile.TemporaryDirectory()
+    if kwargs.get("verbose", False):
+        print(f'Using temporary extension directory {temp_dir.name}...', file=sys.stderr)
+    assert "build_directory" not in kwargs
+    kwargs["build_directory"] = temp_dir.name
+    module = cpp_extension.load_inline(*args, **kwargs)
+    return module, temp_dir
```

Comment:
It's only really scoped if it's a context manager. Otherwise it's just creating a C++ extension with a temporary directory. The options are:
1) change this API to be a context manager. On exit, this will delete the temp_dir.
2) rename the API to be "c++ extension with temporary directory, where the directory will *eventually* clean itself up, we don't know when"

First one is generally more pythonic